x-pack/plugins/lens/public/app_plugin/shared/saved_modal_lazy.tsx

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Suspense, useEffect, useState } from 'react';
+
+import { EuiLoadingSpinner } from '@elastic/eui';
+import { CoreStart } from 'kibana/public';
+import type { SaveModalContainerProps } from '../save_modal_container';
+import type { LensAttributeService } from '../../lens_attribute_service';
+import type { LensPluginStartDependencies } from '../../plugin';
+import type { LensAppServices } from '../types';
+const SaveModal = React.lazy(() => import('../save_modal_container'));
+
+const LensSavedModalLazy = (props: SaveModalContainerProps) => {
+  return (
+    <Suspense fallback={<EuiLoadingSpinner />}>
+      <SaveModal {...props} />
+    </Suspense>
+  );
+};
+
+export function getSaveModalComponent(
+  coreStart: CoreStart,
+  startDependencies: LensPluginStartDependencies,
+  attributeService: () => Promise<LensAttributeService>
+) {
+  return (props: Omit<SaveModalContainerProps, 'lensServices'>) => {
+    const [lensServices, setLensServices] = useState<LensAppServices>();
+
+    useEffect(() => {
+      async function loadLensService() {
+        const { getLensServices } = await import('../../async_services');
+        const lensServicesT = await getLensServices(coreStart, startDependencies, attributeService);
+
+        setLensServices(lensServicesT);
+      }
+      loadLensService();
+    }, []);
+
+    if (!lensServices) {
+      return <EuiLoadingSpinner />;
```

Comment:
Not sure about the loading spinner components here and in line 20 - as the real component is an `EuiModal` it will be rendered absolutely positioned while the loading spinner will be inline which is probably not what the user expects: (note the spinner appearing below the chart - depending on the layout it could lead to jumping content)
![Kapture 2021-06-15 at 11 21 13](https://user-images.githubusercontent.com/1508364/122028020-e50fb880-cdcb-11eb-899a-db5c4325d310.gif)